### PR TITLE
feat: rollup preset

### LINKED ISSUE #22: rollup preset
**Is your feature request related to a problem? Please describe.**
There can be a desync between the cli commands publish and dev-server with respect to rollup configuration.

**Describe the solution you'd like**
We should make a rollup preset.

**Describe alternatives you've considered**
Keep it the same.

### PR DESCRIPTION
**Pull request by crherman7**
## Scope

[closes TICKET-22](https://github.com/crherman7/rechunk/issues/22)

## Implementation

Creates a Rollup preset. This includes all the required rollup plugins to generate a clean bundle that can be used by ReChunk to render a string to a component. It additionally modifies the babel configuration to exclude module resolver plugin and set enableBabelRuntime to false for the react native babel preset.

## Screenshots

<details>
  <summary>ios</summary>

Simulator Screenshot - iPhone 15 - 2024-08-29 at 18 33 51
</details>

<details>
  <summary>android</summary>
  
Screenshot_1724971045
</details>

## How to Test

Run the example app for both platforms and verify the ReChunk components render.

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

### CODE REVIEW

File: src/rollup-preset/index.ts [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+  let babelConfigOptions;
+
+  if (babelConfig?.options) {
+    babelConfigOptions = modifyBabelConfig(
+      babelConfig.options,
+      BABEL_MODIFICATIONS,
+    );
+  }
+
+  // TODO: exclude module-resolver for bundling purposes and add enableBabelRuntime: false to babelConfig
```
Comment (devs9) [inline src/rollup-preset/index.ts]: 🟡  you've already done that

```suggestion

```

File: src/cli/cmds/dev-server.ts [hunk 2]
```diff
@@ -119,40 +102,15 @@ program
...
-                module: 'commonjs',
-                esModuleInterop: true,
-                allowSyntheticDefaultImports: true,
-                downlevelIteration: true,
-              },
-            }),
-          ],
-          logLevel: 'silent',
-        });
+        const rollupBuild = await rollup(await withRechunk({input}));
```
Comment (devs9) [inline src/cli/cmds/dev-server.ts:106]: 🟢 Nice refactor! 

`publish` and `dev-server` commands look much cleaner 👍

File: src/rollup-preset/index.ts [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+ *
+ * @param babelConfig - The loaded Babel configuration.
+ * @param modifications - An object defining plugins/presets to remove or update.
+ * @returns The modified Babel configuration.
+ */
+function modifyBabelConfig(
+  babelConfig: TransformOptions | null,
+  modifications: {
+    plugins?: {
+      [key: string]: null | ((options: PluginOptions) => PluginOptions);
```
Comment (devs9) [inline src/rollup-preset/index.ts]: 🤔 maybe move it in a separate type? occurs several times in this file

Comment (crherman7) [reply src/rollup-preset/index.ts]: yeah good point

File: src/rollup-preset/index.ts (hunk 1 above)
Comment (crherman7) [reply src/rollup-preset/index.ts]: GREAT CATCH!

Comment (crherman7) [reply src/rollup-preset/index.ts]: updated

File: src/rollup-preset/index.ts (hunk 3 above)
Comment (crherman7) [reply src/rollup-preset/index.ts]: updated

**Review by devs9 — APPROVED** [review]
Great job 👍

**Pull request merged**